### PR TITLE
[Small typo] Update match.rst

### PR DESCRIPTION
**Pull request by clementblaudeau**
Some error messages were merged together

**Kind:** documentation

### CODE REVIEW

File: doc/sphinx/language/extensions/match.rst
Comment (Zimmi48) [inline doc/sphinx/language/extensions/match.rst:886]:
```suggestion
   Patterns are incorrect (because constructors are not applied to the correct number of
```

**Review by Zimmi48 — APPROVED** [review]
Great! Thanks for noticing and fixing this. While reading the description, I noticed some incorrect wording. Would you mind applying my suggestion before I merge this PR?

**Comment by clementblaudeau** [comment]
Done ! 
Out of curiosity, if I stumble upon other small typos like this one in the future, is it always worth it to do a PR ? It seems like too much work/attention for you maintainers for such a small change.

**Comment by Zimmi48** [comment]
Thanks!

It's always worth it. It's not a lot of work to review / merge a small PR (and significantly less work than an issue listing typos), but more importantly external contributions are really neat to fix some stuff that we would never fix (or even notice) otherwise.

And fixing typos are not the only contributions from you that would be welcome. If you notice that something is unclear, or could be explained better, or would deserve an example, do not hesitate to, either:
1. open a PR if you see how things could be made better or if you have a proposal to make;
2. open an issue if you want to discuss the content first or would just like to report that you don't understand some part.

I'd like to add that the reference manual was significantly restructured in 8.12 and the consequence is that some new or modified pages may look weird and help is welcome to improve the structure and the content. You can have a look at this [wiki page](https://github.com/coq/coq/wiki/Refman-improvements-in-8.12-and-beyond) to learn more. In particular, if I take the example of the page you edited in this PR, it's the result of merging content coming from two different sources: most of it comes from a page on "Extended pattern matching" that used to be part of the "Addendum" of the reference manual (cf. https://coq.inria.fr/distrib/V8.11.2/refman/) but the first section "Variants and extensions of match" comes from the previous (very long) "Extensions of Gallina" page. This means that the structure of this new page is far from ideal, there might be some redundancies left over, things could be improved by moving around, etc. If that's a part of the doc you're spending time on, do not hesitate to think at how you would structure it so that it would flow better. And if you have ideas of improvements but no time to implement them, you can still share them by opening an issue. This is also welcome.

---

@coqbot: merge now

**Pull request merged**
